engine/collection/ingest/engine.go

```diff
@@ -249,7 +249,10 @@ func (e *Engine) onTransaction(originID flow.Identifier, tx *flow.TransactionBod
 
 	// using the transaction's reference block, determine which cluster we're in.
 	// if we don't know the reference block, we will fail when attempting to query the epoch.
-	refEpoch := refSnapshot.Epochs().Current()
+	refEpoch, err := refSnapshot.Epochs().Current()
+	if err != nil {
+		return fmt.Errorf("could not get current epoch for reference block: %w", err)
+	}
 
```

Comment:
```suggestion
```
I my opinion, the comment above:
https://github.com/onflow/flow-go/blob/6e41e09de9320baa56bcb6aef3119ff97d5606a8/engine/collection/ingest/engine.go#L250-L251
pertains to the entire code block [lines 253-268](https://github.com/onflow/flow-go/blob/6e41e09de9320baa56bcb6aef3119ff97d5606a8/engine/collection/ingest/engine.go#L253-L268) ... so I would be inclined to not break this block up (14 lines only), since its all dealing with the same gaol of getting the cluster